Carlos muniz/symmath removal unrevert

Carlos-Muniz opened the pull request:
## Description

This PR un-reverts #29909 and [adds a fix](https://github.com/openedx/edx-platform/pull/29869#issuecomment-1036397817) to the issue that [broke the edxapp build on GoCD](https://github.com/openedx/edx-platform/pull/29869#issuecomment-1035649004).

## Testing Instructions
The code has been tested:
- Manually: 
  - A custom dev image of lms/cms was built using this branch via tutor
  - Tutor was started in dev mode using these images
  - While these containers were running, the version of `openedx-calc` was checked to be `3.0.1`, and that `symmath` was not in `common/lib/`
  - In the CMS GUI (Studio), a new course was formed, and a new symbolic math problem was initiated.
  - Answers that were mathematically correct, but symbolically different were entered into the sample problem
- Locally: Pytest via Tutor
- Online: All checks have passed

## Other information

For more information see #29909 and #29869

Commit:
Revert "Merge pull request #29909 from openedx/revert-29869-Carlos-Muniz/symmath-removal"

This reverts commit 8c0db8ddff9760ad78b3122113d718b21e2bef04, reversing
changes made to 1156c62014f275613ed0b02d70677dbdfc605c25.

Commit:
fix: Remove misplaced `-e`

`-e` was wrongfully placed in front of `common/lib/sandbox-packages`,
which may have most likely been causing the edxAPP to break.

kdmccormick commented:
@Carlos-Muniz if you run `make upgrade` again, does the `- e` come back?

kdmccormick commented:
pro tip @Carlos-Muniz : if you don't want to run make upgrade locally, you can go to Actions > Upgrade Requirements and run the action against your brach, which'll open a new PR with the results.

kdmccormick commented on scripts/post-pip-compile.sh:
```diff
@@ -17,7 +17,7 @@ function clean_file {
     # change absolute paths for local editable packages back to relative ones
     FILE_CONTENT=$(<${FILE_PATH})
-    FILE_URL_REGEX="-e (file:///[^"$'\n'"]*)/common/lib/symmath"
+    FILE_URL_REGEX="-e (file:///[^"$'\n'"]*)/common/lib/xmodule"
```
@Carlos-Muniz I think the problem is that `py38.txt` (the file that ended up with the extra `-e file://...` in it) doesn't have include `common/lib/xmodule`, so it didn't match this regex and therefore never got its line for `common/lib/sandbox-packages` fixed.

What if we changed the regex to match any `common/lib/<packagename>` path? Something like:
```
FILE_URL_REGEX="-e (file:///[^"$'\n'"]*)/common/lib/\w+"
```

Carlos-Muniz replied:
I think you're right.
And I think https://github.com/openedx/edx-platform/pull/29918 proves it

kdmccormick replied:
Nice!

Commit:
fix: Change regex to apply to right dirs

`py38.txt` does not include `common/lib/xmodule` so it doesn't match the
regex. Therefor, it never got its lines in `common/lib/sandbox-packages`
fixed. If we change the regex to match any `common/lib/<packagename>` it
should work correctly.

kdmccormick reviewed (approved):
:shipit: :shipit: :shipit: :shipit: :shipit:
